### PR TITLE
Overhaul team mapping

### PR DESCRIPTION
**Pull request by arnavxkohli**
These changes deal with converting the raw team type into team pointers. Currently there is a bug caused by there not being any logic to declare a winner in case only one agent/one team is left. A work-around is used in the code, which otherwise runs fine:

```go
if team == nil {
    continue
}
```

### CODE REVIEW

**Comment by arnavxkohli** [comment]
Would be nice if the people working on common pool updation can test their code with these updates before merging @kishan-v @nlewxxs

File: server/EnvironmentServer.go [hunk 1]
```diff
@@ -334,7 +348,7 @@ func (cs *EnvironmentServer) CreateAndInitTeamWithAgents(agentIDs []uuid.UUID) u
 }
 
 // agent get team
-func (cs *EnvironmentServer) GetTeam(agentID uuid.UUID) common.Team {
+func (cs *EnvironmentServer) GetTeam(agentID uuid.UUID) *common.Team {
```
Comment (nlewxxs) [inline server/EnvironmentServer.go:351]: Are you not getting this error when you're running the code?
```
panic: interface conversion: *environmentServer.EnvironmentServer is not common.IServer: missing method GetTeam
```

You need to change the return type of: 

```go
GetTeam(agentID uuid.UUID) Team
```
to `*Team` in `IServer.go`. 

please run code before pushing :((

**Review by nlewxxs — CHANGES_REQUESTED** [review]
The common pools are indeed not working properly at the minute, probably a small change that needs to be made somewhere given the switch to pointers in team. 

Will take a look later tonight, unless @kishan-v gets there first.

File: server/EnvironmentServer.go (hunk 1 above)
Comment (kishan-v) [reply server/EnvironmentServer.go:351]: +1 
@nlewxxs is spot on here.

Comment (arnavxkohli) [reply server/EnvironmentServer.go:351]: image

**Comment by kishan-v** [comment]
Fixed the bug so common pool increases/decreases as expected now

**Review by kishan-v — APPROVED** [review]

**Review by Somewheref — APPROVED** [review]
nice stuff

**Review by nlewxxs — APPROVED** [review]
Is there a reason we don't need `UpdateCommonPools()` anymore? I'm on my phone atm its hard to tell.

**Comment by kishan-v** [comment]
> Is there a reason we don't need `UpdateCommonPools()` anymore? I'm on my phone atm its hard to tell.

I've re-added that behaviour of only updating the common pool after all agents have contributed/withdrawn in #31 . Please let me know what you think.

#31 would need to be merged into this PR (#28 ) before merging to main.

**Comment by kishan-v** [comment]
There are now merge conflicts to main though :(

**Pull request merged**